modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -0,0 +1,493 @@
+										indeterminate={isSelectAllIndeterminate()}
+										onChange={({target: {checked}}) =>
+											setFields(
+												fields.map((field) => ({
+													...field,
+													selected: checked,
+												}))
+											)
+										}
+									/>
+								</ManagementToolbar.Item>
+
+								<ManagementToolbar.Item className="nav-item-expand">
+									<ClayInput.Group>
+										<ClayInput.GroupItem>
+											<ClayInput
+												insetAfter
+												onChange={(event) =>
+													onSearch(event.target.value)
+												}
+												placeholder={Liferay.Language.get(
+													'search'
+												)}
+												type="text"
+												value={query}
+											/>
+
+											<ClayInput.GroupInsetItem
+												after
+												tag="span"
+											>
+												<ClayButtonWithIcon
+													aria-label={Liferay.Language.get(
+														'search'
+													)}
+													displayType="unstyled"
+													symbol="search"
+												/>
+											</ClayInput.GroupInsetItem>
+										</ClayInput.GroupItem>
+									</ClayInput.Group>
+								</ManagementToolbar.Item>
+							</ManagementToolbar.ItemList>
+						</ManagementToolbar.Container>
+
+						<div className="fields pb-2 pt-2">
+							{visibleFields.length ? (
+								visibleFields.map(({name, selected}) => (
+									<div
+										className="pb-2 pl-3 pr-3 pt-2"
+										key={name}
+									>
+										<ClayCheckbox
+											checked={selected}
+											label={name}
+											onChange={({target: {checked}}) => {
+												setFields(
+													fields.map((field) =>
+														field.name === name
+															? {
+																	...field,
+																	selected: checked,
+															  }
+															: field
+													)
+												);
+											}}
+										/>
+									</div>
+								))
+							) : (
+								<div className="pb-2 pl-3 pr-3 pt-2 text-3">
+									{Liferay.Language.get('no-results-found')}
+								</div>
+							)}
+						</div>
+					</>
+				)}
+			</ClayModal.Body>
+
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton
+							onClick={() => {
+								closeModal();
+
+								onSave();
+							}}
+						>
+							{Liferay.Language.get('save')}
+						</ClayButton>
+
+						<ClayButton
+							displayType="secondary"
+							onClick={() => closeModal()}
+						>
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+					</ClayButton.Group>
+				}
+			/>
+		</div>
+	);
+};
+
+const Fields = ({fdsView, fdsViewsURL}: FDSViewSectionInterface) => {
+	const [fdsFields, setFDSFields] = useState<Array<FDSFieldType> | null>(
+		null
+	);
+
+	const fdsFieldsOrderRef = useRef('');
+
+	const getFDSFields = async () => {
+		const response = await fetch(
+			`${API_URL.FDS_FIELDS}?filter=(${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_FIELD_ID} eq '${fdsView.id}')&nestedFields=${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_FIELD}`
```

Comment:
What's this `eq`?

Reply:
Equals, [odata syntax for filter](https://help.liferay.com/hc/en-us/articles/360036343152-Filter-Sort-and-Search). A way to should fields only from currently selected FDS View.